### ISSUE TITLE
How to avoid multiple triggers when modal.keepMounted is true

### ISSUE THREAD

**Issue by ficapy** [eBay/nice-modal-react#35]
Thanks a lot for such a great library that makes the code better split!

I wanted to keep the user's temporary input and found that `modal.keepMounted = true ` would do it

but this will cause the callback function to be triggered multiple times on the last submit, and I want it to be called only once at the end.

[code Link](https://codesandbox.io/s/happy-haze-8msk62?file=/src/App.tsx)

**Comment by supnate** [comment]
You are welcome @ficapy !

I can't reproduce the issue. When I click submit only one log message in the console.

BTW, we usually use code like below to set `keepMounted`:

```js
const MyModal = useModal(MyAntdModal, { keepMounted: true });
```

**Comment by ficapy** [comment]
@supnate Hello, I recorded a video
https://www.loom.com/share/8898a1e16af94ca4b579222648f391a8

```ts
import { Modal, Button, Form } from "antd";
import FormBuilder from "antd-form-builder";
import NiceModal, { useModal, antdModal } from "@ebay/nice-modal-react";

const MyAntdModal = NiceModal.create(() => {
  const modal = useModal();
  const [form] = Form.useForm();

  const handleSubmit = async () => {
    await form.validateFields();
    const val = form.getFieldsValue();
    modal.resolve(val);
  };
  const meta = [{ key: "name", label: "Name", required: true }];
  return (
    <Modal title="Hello Antd" {...antdModal(modal)} onOk={handleSubmit}>
      <Form form={form}>
        <FormBuilder meta={meta} form={form} />
      </Form>
    </Modal>
  );
});

export default function App() {
  const MyModal = useModal(MyAntdModal, { keepMounted: true });
  const handleOk = async () => {
    const val = await MyModal.show();
    // output multiple times
    console.log(val);
    await MyModal.hide();
  };
  return (
    <Button type="primary" onClick={handleOk}>
      Show Modal
    </Button>
  );
}

```

**Comment by ficapy** [comment]
I read the source code and the cause is similar to the following code, how do I avoid triggering then multiple times?
```js
let theResolve 
const a = new Promise((resolve) => {
    theResolve = resolve
})
a.then((v) => {console.log(v)})
a.then((v) => {console.log(v)})
a.then((v) => {console.log(v)})

theResolve('hello')
```

**Comment by supnate** [comment]
Fixed, plz update to v1.2.1. Thanks again for reporting it!
